city_scrapers/spiders/il_procurement_policy.py

```diff
@@ -0,0 +1,153 @@
+import re
+from datetime import datetime, time
+
+from city_scrapers_core.constants import BOARD
+from city_scrapers_core.items import Meeting
+from city_scrapers_core.spiders import CityScrapersSpider
+
+
+class IlProcurementPolicySpider(CityScrapersSpider):
+    name = "il_procurement_policy"
+    agency = "Illinois Procurement Policy Board"
+    timezone = "America/Chicago"
+    start_urls = [
+        'https://www2.illinois.gov/sites/ppb/Pages/future_board_minutes.aspx',
+        'https://www2.illinois.gov/sites/ppb/Pages/board_minutes.aspx'
+    ]
+
+    def parse(self, response):
+        """
+        `parse` should always `yield` Meeting items.
+
+        Change the `_parse_title`, `_parse_start`, etc methods to fit your scraping
+        needs.
+        """
+        if 'future' in response.url:
+            yield from self._upcoming_meetings(response)
+        else:
+            yield from self._prev_meetings(response)
+
+    def _parse_title(self, item):
+        """Parse or generate meeting title."""
+        title_str = item['title'].split()
+        if not len(title_str):
+            return ""
+        name_str = title_str[len(title_str) - 1] + " Board Meeting Minutes"
+        return name_str
+
+    def _parse_description(self, item):
+        """Parse or generate meeting description."""
+        return ""
+
+    def _parse_classification(self, item):
+        """Parse or generate classification from allowed options."""
+        return BOARD
+
+    def _parse_start(self, item):
+        """Parse start datetime as a naive datetime object."""
+        time_object = time(10, 0)
+        date_str = " ".join(item.css("*::text").extract()).strip()
+        date_str = re.sub("Agenda.pdf", "", date_str).strip()
+        date_object = datetime.strptime(date_str, "%B %d, %Y").date()
+        return datetime.combine(date_object, time_object)
+
+    def _parse_links(self, item, response):
+        """Parse or generate links."""
+        links = []
+        title_str = " ".join(item.css("*::text").extract()).strip()
+        if 'pdf' in title_str:
+            title_str = re.sub("Agenda.pdf", "", title_str).strip()
+            title_str += " Agenda"
+        links.append({
+            'title': title_str,
+            'href': response.urljoin(item.attrib['href']),
+        })
+        return links
+
+    def _parse_past_links(self, response):
+        """parse start datetime as a naive datetime object"""
```

Comment:
It looks like there's a lot of logic going into parsing out pieces of text that we don't end up using here. Since it seems like dates can be repeated, using a `defaultdict` to store start dates associated with a list of links like we do in this example might be easier:

https://github.com/City-Bureau/city-scrapers/blob/20a12ba5d76186cba65b45f7f764f02393d4a991/city_scrapers/spiders/chi_il_medical_district.py#L27-L38

Let me know if that makes sense, but it should simplify things here a good amount

Reply:
My general understanding is that I parse out the dates and map it to each link?